File: src/ChpokkWeb/SystemFiles/Templates/ProjectTemplates/CSharp/Silverlight/1041/BusinessApplication/Helpers/NotOperatorValueConverter.cs
namespace $safeprojectname$
{
    using System;
    using System.Windows.Data;

    /// <summary>
    /// boolean プロパティの逆数を依存関係プロパティにバインドできる双方向の <see cref="IValueConverter"/> です。
    /// </summary>
    public class NotOperatorValueConverter : IValueConverter
    {
        /// <summary>
        /// 指定された <paramref name="value"/> を逆数に変換します。
        /// </summary>
        /// <param name="value">変換する <c>bool</c> 値です。</param>
        /// <param name="targetType">変換先の型です (無視)。</param>
        /// <param name="parameter">省略可能なパラメーターです (無視)。</param>
        /// <param name="culture">変換のカルチャです (無視)。</param>
        /// <returns>入力 <paramref name="value"/> の逆数です。</returns>
        public object Convert(object value, Type targetType, object parameter, System.Globalization.CultureInfo culture)
        {
            return !((bool)value);
        }

        /// <summary>
        /// <see cref="Convert"/> の逆数です。
        /// </summary>
        /// <param name="value">変換によって元に戻された後の値です。</param>
        /// <param name="targetType">変換先の型です (無視)。</param>
        /// <param name="parameter">省略可能なパラメーターです (無視)。</param>
        /// <param name="culture">変換のカルチャです (無視)。</param>
        /// <returns>入力 <paramref name="value"/> の逆数です。</returns>
        public object ConvertBack(object value, Type targetType, object parameter, System.Globalization.CultureInfo culture)
        {
            return !((bool)value);
        }
    }
}

File: src/ChpokkWeb/SystemFiles/Templates/ProjectTemplates/VisualBasic/Silverlight/1042/BusinessApplication/Helpers/NotOperatorValueConverter.vb
Imports System
Imports System.Windows.Data

''' <summary>
''' boolean 속성의 역을 종속성 속성에 바인딩할 수 있게 해주는 양방향 <see cref="IValueConverter"/>입니다.
''' </summary>
Public Class NotOperatorValueConverter
    Implements IValueConverter


    ''' <summary>
    ''' 지정된 <paramref name="value"/>를 역으로 변환합니다.
    ''' </summary>
    ''' <param name="value">변환할 <c>Boolean</c> 값입니다.</param>
    ''' <param name="targetType">변환할 type입니다(무시됨).</param>
    ''' <param name="parameter">선택적 매개 변수입니다(무시됨).</param>
    ''' <param name="culture">변환의 문화권입니다(무시됨).</param>
    ''' <returns>입력 <paramref name="value"/>의 역입니다.</returns>
    Public Function Convert(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As System.Globalization.CultureInfo) As Object Implements IValueConverter.Convert
        Return Not DirectCast(value, Boolean)
    End Function

    ''' <summary>
    ''' <see cref="Convert"/>의 역입니다.
    ''' </summary>
    ''' <param name="value">다시 변환할 값입니다.</param>
    ''' <param name="targetType">변환할 type입니다(무시됨).</param>
    ''' <param name="parameter">선택적 매개 변수입니다(무시됨).</param>
    ''' <param name="culture">변환의 문화권입니다(무시됨).</param>
    ''' <returns>입력 <paramref name="value"/>의 역입니다.</returns>
    Public Function ConvertBack(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As System.Globalization.CultureInfo) As Object Implements IValueConverter.ConvertBack
        Return Not DirectCast(value, Boolean)
    End Function
End Class